lib/ocamlorg_web/graphql.ml

```diff
@@ -271,25 +329,49 @@ let schema t : Dream.request Graphql_lwt.Schema.schema =
                   "limit"
                   ~typ:int
               ]
-          ~resolve:(fun _ () startswith offset limit ->
-            let packages = Package.all_packages_latest t in
-            let total_packages = List.length packages in
-            let limit =
-              match limit with None -> total_packages | Some limit -> limit
-            in
-            get_packages_result total_packages offset limit startswith packages)
-      ; field
+          ~resolve:(fun _ () contains offset limit ->
+            let all_packages = Package.all_packages_latest t in
```

Comment:
Could this maybe be done inside the `all_packages_latest` function ? I don't mind if it isn't too much it just seems odd to me that `all_packages_result` requires both the state `t` and a list of all of the packages?